Source/JavaScriptCore/runtime/ArrayBuffer.h

```diff
@@ -320,6 +321,8 @@ class ArrayBuffer final : public GCIncomingRefCounted<ArrayBuffer> {
     Expected<int64_t, GrowFailReason> grow(VM&, size_t newByteLength);
     Expected<int64_t, GrowFailReason> resize(VM&, size_t newByteLength);
 
+    Vector<uint8_t> vector() const { return { static_cast<const uint8_t*>(data()), byteLength() }; };
```

Comment:
cc @Constellation to review the ArrayBuffer changes.

Reply:
Let's name it `toVector()`. The name `vector()` makes me think that it is a getter returning a `m_vector` field.
Also, new data is allocated and copied. Is it expected? Maybe copying is fine. If you would like to use direct pointer and size, then you need to be extra careful since ArrayBuffer is resizable and growable. If you invoke some JS operations, then pointer can be invalidated (and byteLength() can be changed concurrently). So, maybe copying is fine here for safety.

Reply:
I hadn't considered the size changes, but given that concern I agree that making a copy sounds like a better idea.